tools/autograd/gen_python_functions.py

```diff
@@ -320,6 +320,19 @@ def gen(
         fm, functions, lambda fn: True, "python_return_types.cpp"
     )
 
+    valid_tags = parse_tags_yaml(tags_yaml_path)
+
+    def gen_tags_enum() -> Dict[str, str]:
+        return {
+            "enum_of_valid_tags": (
+                'py::enum_<Tags>(m, "Tags")'
```

Comment:
Why not have all the static part directly in the template? That will remove the need for a lot of `\n` and linebreaks here.